### PR TITLE
chore(stepfunctions-tasks): clean up aws-stepfunctions-tasks linter exclusions

### PR DESCRIPTION
**Pull request by shivlaks**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts [hunk 1]
```diff
@@ -7,36 +7,44 @@ import * as s3 from '@aws-cdk/aws-s3';
 import * as sfn from '@aws-cdk/aws-stepfunctions';
 import { Construct, Duration } from '@aws-cdk/core';
 
+/**
+ * Task to train a machine learning model using Amazon SageMaker
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: `@experimental` as well?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts [hunk 2]
```diff
@@ -55,31 +63,43 @@ export interface Channel {
...
-     * Content type
+     * The MIME type of the data.
+     *
+     * @default - None
      */
     readonly contentType?: string;
 
     /**
-     * Location of the data channel
+     * Location of the channel data.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: It should be `data channel`, no?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts [hunk 3]
```diff
@@ -151,16 +182,21 @@ export interface OutputDataConfig {
...
+ * When the job reaches the time limit, Amazon SageMaker ends the training job.
+ *
  * @experimental
  */
 export interface StoppingCondition {
     /**
      * The maximum length of time, in seconds, that the training or compilation job can run.
      */
-    readonly maxRuntime?: Duration;
+    readonly maxRuntime: Duration;
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: By any chance, is this a breaking change? Should we announce it as a breaking change?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts [hunk 4]
```diff
@@ -193,17 +231,20 @@ export interface ResourceConfig {
...
 
 /**
+ * Specifies the VPC that you want your Amazon SageMaker training job to connect to.
  *
  * @experimental
  */
 export interface VpcConfig {
     /**
-     * VPC id
+     * VPC ID
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: It's not the VPC id. It's the vpc itself.

Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: will remove the id

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts (hunk 3 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: this was unintended... thought I'd removed it... good catch.
default is an hour, will update

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts (hunk 2 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: I pulled it from the SageMaker [documentation](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_Channel.html)

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts]: good call, adding it

**Pull request merged**